### PR TITLE
Order Status Mark II: add Network changes

### LINKED ISSUE #717: Storage: unused Order core data property needs deleted
The `Order+CoreDataProperties.swift` file did not get regenerated after my last set of code review changes made to PR #704. The data model and mapping model **did** get regenerated, so the extra `OrderStatus` relationship found in `Order+CoreDataProperties` is not referenced anywhere else or used in any code.

To Do: delete this line of code: https://github.com/woocommerce/woocommerce-ios/blob/develop/Storage/Storage/Model/Order%2BCoreDataProperties.swift#L56

### PR DESCRIPTION
**Pull request by mindgraffiti**
Ref. #262. Fixes #717.

This PR is getting long. I'm cutting it here. tl:dr: Work done in Storage, Networking, and Yosemite.

In Storage, changes include: siteID and total are added to the OrderStatus entity. Use the siteID and the slug to load individual OrderStatuses. The OrderStatus+CoreDataProperties has been regenerated. There is no new mapping model because the changes are lightweight.

In Network: The ReportOrderTotalsMapper has been modified to map JSON to the OrderStatus model. This model supports all the fields returned by the API response, and adds on the siteID. In the ReportRemote, the loadTotals: function has been decoupled. 

## To Test
- [x] Make sure all unit tests are green!
- [x] Run the app and make sure Order Totals in the Dashboard continue to load correctly

cc: @ctarda 

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                return
+            }
+
+            self?.upsertStoredOrderStatuses(siteID: siteID, readOnlyOrderStatuses: orderStatuses)
+            onCompletion(orderStatuses, nil)
+        }
+    }
+}
+
+
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: Hound needs our styles added to it. 2 lines of vertical whitespace following a pragma mark.

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import Foundation
+import Networking
+import Storage
+import CocoaLumberjack
+
+
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: Hound needs our styling rules added to it. Two lines of vertical whitespace following a pragma mark.

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (bummytime) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: @mindgraffiti 💯

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            } else {
+                let newStorageItem = storage.insertNewObject(ofType: Storage.OrderStatus.self)
+                newStorageItem.update(with: readOnlyItem)
+            }
+        }
+
+        // Now, remove any objects that exist in storageOrderStatuses but not in readOnlyOrderStatuses
+        if let storageOrderStatuses = storage.loadOrderStatuses(siteID: siteID) {
+            storageOrderStatuses.forEach({ storageItem in
+                if readOnlyOrderStatuses.first(where: { $0.slug == storageItem.slug } ) == nil {
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: This is part of our styling. It's still consistent and readable.

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (bummytime) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: @mindgraffiti Added a PR to address this in #725

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (bummytime) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: @mindgraffiti TBH, I am not opposed to leaving this rule active...admittedly I don't care either way. Grep'ing through the code, it appears we would only need to fix three(ish) spots (all of which is my code 😆 ). Any strong opinions with that? @ctarda ?

File: Networking/NetworkingTests/Mapper/ReportOrderMapperTests.swift [hunk 4]
```diff
@@ -16,17 +20,71 @@ class ReportOrderMapperTests: XCTestCase {
...
-        XCTAssertEqual(reportTotals?[OrderStatusKey(rawValue: "cia-investigation")], 10)
+        XCTAssertEqual(reportTotals.count, 9)
+        XCTAssertEqual(reportTotals[.pending], 123)
+        XCTAssertEqual(reportTotals[.processing], 4)
+        XCTAssertEqual(reportTotals[.onHold], 5)
+        XCTAssertEqual(reportTotals[.completed], 6)
+        XCTAssertEqual(reportTotals[.cancelled], 7)
+        XCTAssertEqual(reportTotals[.refunded], 8)
+        XCTAssertEqual(reportTotals[.failed], 9)
+        XCTAssertEqual(reportTotals[OrderStatusKey(rawValue: "cia-investigation")], 10)
```
Comment (astralbodies) [inline Networking/NetworkingTests/Mapper/ReportOrderMapperTests.swift]: 🤣

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: Thank you @bummytime!

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: I mis-read that. Hound is correct on this one. I thought it was complaining about the space between the inner-most curly braces. It's complaining that the spacing is inconsistent between the curly brace and closing parens. Nevermind @bummytime, I will fix.

File: Storage/Storage/Tools/StorageType+Extensions.swift [hunk 5]
```diff
@@ -82,6 +82,21 @@ public extension StorageType {
         return firstObject(ofType: OrderStatsItem.self, matching: predicate)
     }
 
+    /// Retrieves all of the Stores OrderStatuses for the provided siteID.
+    ///
+    public func loadOrderStatuses(siteID: Int) -> [OrderStatus]? {
```
Comment (bummytime) [inline Storage/Storage/Tools/StorageType+Extensions.swift]: @mindgraffiti I think this should be `loadOrderStatii()` 🤣🤣🤣

JK — I never get to correct your grammar/typos, so -

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+
+// MARK: - Persistence
+//
+extension OrderStatusStore {
+
+    /// Updates (OR Inserts) the specified ReadOnly OrderStatus Entities into the Storage Layer.
+    ///
+    func upsertStoredOrderStatuses(siteID: Int, readOnlyOrderStatuses: [Networking.OrderStatus]) {
```
Comment (bummytime) [inline Yosemite/Yosemite/Stores/OrderStatusStore.swift]: We could probably move the work in this func to a background thread via derivedStorage, but for the purposes of this PR it looks good.

**Comment by mindgraffiti** [comment]
Thanks @astralbodies @bummytime !

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: Storage/Storage/Tools/StorageType+Extensions.swift (hunk 5 above)
Comment (mindgraffiti) [reply Storage/Storage/Tools/StorageType+Extensions.swift]: 🤔 I think you have me there!

File: Yosemite/Yosemite/Stores/OrderStatusStore.swift (hunk 6 above)
Comment (mindgraffiti) [reply Yosemite/Yosemite/Stores/OrderStatusStore.swift]: I'll do that in my Mark III

**Pull request merged**